### ISSUE TITLE
Is PumpNet currently offline?

### ISSUE THREAD

**Issue by Enh4** [pumpitupdev/pumpnet-feedback#5]
Hi! Enh4 again! 

Im trying to play some NXA right now, but seems like the game get stuck in the USB screen... I also tried not using the usb and it works properly.

Just checking if the server is offline.
Thanks!

**Comment by voidderef** [comment]
Thanks for reaching out. I took a look and everything appeared fine and operational to me. However, just in case something was stuck that I couldn't see, I restarted the service. Please give it another shot and let me know.

**Comment by Enh4** [comment]
Hi! Thanks for the response! 

> [M][2023/12/26-19:16:31:659][patch-usb-unlock][usb-unlock.c:46] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> ERROR: ld.so: object '/home/andamiro/Desktop/piu/21_NXA/hook.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
> [M][2023/12/26-19:16:31:680][patch-net-profile][net-profile.c:491] open /mnt/0/nxasave.bin 0 0
> [I][2023/12/26-19:16:31:680][patch-net-profile][net-profile.c:433] Found pumpnet profile file /mnt/0/pumpnet.bin
> [M][2023/12/26-19:16:31:681][patch-net-profile][net-profile.c:372] Setup virtual file, player 0, file_type 0
> [M][2023/12/26-19:16:31:681][patch-net-profile][net-profile.c:392] Setup virtual file done, size 90868
> [I][2023/12/26-19:16:31:681][patch-net-profile][net-profile.c:463] Profile file player 0, file_type 0 downloading from server...
> [I][2023/12/26-19:16:31:681][pumpnet][pumpnet.c:190] [71535570308CEA68] Get save
> [M][2023/12/26-19:16:31:681][http][http.c:245] [71535570308CEA68][https://pumpnet/pumpnet/usbprofile/v1/nxa/save] GET 24 90872

Here is where I get stuck... Dont know what could be wrong...

This also happens in NX2

> [M][2023/12/26-19:33:14:769][sec-microdog40][microdog40.c:318] SEC_MICRODOG40_REQ_OPCODE_CONVERT
> [M][2023/12/26-19:33:14:769][sec-microdog40][microdog40.c:249] Converting: C9 67 CD E6 AE AF F8 49 18 31 D0 7B CE A2 D3 42  ==> 7534772C
> [M][2023/12/26-19:33:14:769][sec-microdog40][microdog40.c:416] Encrypting response...
> [M][2023/12/26-19:33:14:769][sec-microdog40][microdog40.c:424] Transaction successful
> [M][2023/12/26-19:33:16:474][redir.c][redir.c:363] Redirect /proc/bus/usb/devices -> /sys/kernel/debug/usb/devices
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:31] Spoofing VID 0x058f for usb stick
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:31] Spoofing VID 0x058f for usb stick
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:31] Spoofing VID 0x058f for usb stick
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:31] Spoofing VID 0x058f for usb stick
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:474][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:31] Spoofing VID 0x058f for usb stick
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:31] Spoofing VID 0x058f for usb stick
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:31] Spoofing VID 0x058f for usb stick
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> [M][2023/12/26-19:33:16:475][patch-usb-unlock][usb-unlock.c:44] Spoofing serial number of usb stick to kill serial number locking of profile in a moment...
> ERROR: ld.so: object '/home/andamiro/Desktop/piu/20_NX2/hook.so' from LD_PRELOAD cannot be preloaded (wrong ELF class: ELFCLASS32): ignored.
> [M][2023/12/26-19:33:16:496][patch-net-profile][net-profile.c:444] open /mnt/0/nx2save.bin 0 0
> [I][2023/12/26-19:33:16:496][patch-net-profile][net-profile.c:393] Found pumpnet profile file /mnt/0/pumpnet.bin
> [M][2023/12/26-19:33:16:496][patch-net-profile][net-profile.c:338] Setup virtual file, player 0, file_type 0
> [M][2023/12/26-19:33:16:496][patch-net-profile][net-profile.c:355] Setup virtual file done, size 30780
> [I][2023/12/26-19:33:16:496][patch-net-profile][net-profile.c:420] Profile file player 0, file_type 0, refId 3674AFD4C18989B3 downloading from server...
> [I][2023/12/26-19:33:16:497][pumpnet][pumpnet.c:172] [4737304702427CC7][3674AFD4C18989B3] Get save
> [M][2023/12/26-19:33:16:497][http][http.c:227] [4737304702427CC7][https://pumpnet/pumpnet/usbprofile/v1/nx2/save] GET 24 30784

**Comment by icex2** [comment]
That's odd, the log output shows that the client apparently is able to successfully get the data:

```
[M][2023/12/26-19:33:16:497][http][http.c:227] [4737304702427CC7][https://pumpnet/pumpnet/usbprofile/v1/nx2/save] GET 24 30784
```

and

```
[M][2023/12/26-19:16:31:681][http][http.c:245] [71535570308CEA68][https://pumpnet/pumpnet/usbprofile/v1/nxa/save] GET 24 90872
```

Have you tried playing the game with a local profile instead, so without pumpnet but still with a usb to get the save and rank files saved on your usb?

**Comment by Enh4** [comment]
> That's odd, the log output shows that the client apparently is able to successfully get the data:
> 
> ```
> [M][2023/12/26-19:33:16:497][http][http.c:227] [4737304702427CC7][https://pumpnet/pumpnet/usbprofile/v1/nx2/save] GET 24 30784
> ```
> 
> and
> 
> ```
> [M][2023/12/26-19:16:31:681][http][http.c:245] [71535570308CEA68][https://pumpnet/pumpnet/usbprofile/v1/nxa/save] GET 24 90872
> ```
> 
> Have you tried playing the game with a local profile instead, so without pumpnet but still with a usb to get the save and rank files saved on your usb?

Yes I'm actually playing now with an old save I had (the one I gave you back in the testing days) and I found it does work and it was from Agoust 2009, so it has the 4 secret songs and one item that gives you 30 extra secs on World Max.

The only issue I have when using offline profile is that sometimes the screen freezes for a second or so (you can still hear the music in the background tho, and sometimes if I hit a step while this happens, it might take the input and wont count it as a miss). This happens in every mode but mainly on World Max and Brain Shower, and it happens the moment music starts, sometimes even in the middle of the song.

This issue doesn't happen when using pumpnet profile.

But in this moment, for some reason, pumpnet isn't working. I haven't change anything from the game since the last time I successfully connected to the server.

**Comment by voidderef** [comment]
> The only issue I have when using offline profile is that sometimes the screen freezes for a second or so (you can still hear the music in the background tho, and sometimes if I hit a step while this happens, it might take the input and wont count it as a miss). This happens in every mode but mainly on World Max and Brain Shower, and it happens the moment music starts, sometimes even in the middle of the song.

That's quite a bad issue and I don't know why and how offline profiles have that issue but online ones don't have it. I would have expected something like that rather the other way around.

> But in this moment, for some reason, pumpnet isn't working. I haven't change anything from the game since the last time I successfully connected to the server.

I checked the server again and couldn't spot any issues easily, unfortunately. My capacity these days is unfortunately very limited. As this issue seems to require more effort than expected, I unfortunately have to push it back to "some time in the future". Sorry for the disappointing estimate and outlook. There is also nobody else who could pick this up regarding maintenance today.

**Comment by Enh4** [comment]
> That's quite a bad issue and I don't know why and how offline profiles have that issue but online ones don't have it. I would have expected something like that rather the other way around.

Yes! I'm not sure (for both issues), if I'm the only one having these problems, or if it's something general.

I'm running all the mixes directly from a lubuntu distro, which I don't know if that's the best way for the game to be played. (I'm really dumb with linux, code, consoles,etc. And this is the easiest way I found to run all the games in the same hdd). My specs are powerful (Ryzen 9 3900x, RTX2080 Super, and 16GB RAM) but I also don't know if this is something bad for old games like these... (even tho they all work flawlessly with exception of NX2 and NXA which have these freezing issues with offline profiles, or also random sudden test menu appearing while playing every now and then.)

When it comes to pumpnet servers, I've always been able to play... This is only the second time having issues, and it's the same one that happen a couple of months ago, which it was solved back then. So if you are saying that there's nothing wrong (I don't know if you were able to test by your own), then I think I could be the problem... but is weird since I didn't change anything in my setup. (nobody else is having this same issue?)

> My capacity these days is unfortunately very limited. As this issue seems to require more effort than expected, I unfortunately have to push it back to "some time in the future". Sorry for the disappointing estimate and outlook. There is also nobody else who could pick this up regarding maintenance today.

Don't worry, I totally appreciate all the effort you made bringing all these mixes playable once again. I wish I was able to help with code and all that stuff but I'm useless when it comes to it.

I will keep playing with my old NXA profile, even with the freezing I could keep progressing on world max somehow! 
Please, let me know if you need anything, or if you have some news about it!
